test/mitmproxy/contentviews/test_image_parser.py

```diff
@@ -0,0 +1,88 @@
+import io
+
+from mitmproxy.contentviews.image import image_parser
+from mitmproxy.test import tutils
+
+class TestPngParser:
+    def test_png_size(self):
```

Comment:
This looks way better now! We can make @Kriechi happy and parameterize this a bit:


```python
@pytest.mark.parametrize("filename,metadata", {
    "ct1n0g04.png": [
        ('Format', 'Portable network graphics'),
        ('Size', '32 x 32 px'),
        ('gamma', '1.0')
    ],
    "ctzn0g04.png": # ...
}.items())
def test_parse_png(filename, metadata):
    with open(tutils.test_data.path(filename), "rb") as f:
         assert metadata == image_parser.parse_png(f.read())
```

No need to to the `in` checks, let's just specify them entirely in the correct order right away.